### DISCUSSION TITLE
New features

### DISCUSSION THREAD

**Discussion by demo-exe** [alt-romes/programmer-calculator#72] [Ideas]
Hi!
I like this project and came up with some features I might implement, but before that I'd like to see they make any sense to you.
Please tell me if you'd like to see any of:
1. tests rewritten to pytest - current sh implementation is faster sure but I find it hard to read (or write)
2. `80h` hex notation and `b01010` binary notation
3. octal numbers
4. ** as exponent
5. `ans` keyword as current accumulator value 
6. in colored mode 0s and 1s denoted with different color

**Comment by alt-romes** [comment]
Hi, thanks!

I'll reply in order

1. I don't think we should introduce another testing "framework". Adding tests in python would definitely improve the tests overall, but it would also add a dependency on python for contributors. I believe the tests we have serve their main purpose, so I say we leave them be for now
2. Regarding number base format, I don't immediatly think we should introduce new syntax for hexadecimal and binary. The current choice, `0x` and `0b`, follow C's syntax for hex and binary. But I can be convinced otherwise here, if you really think also allowing a different syntax is better
3. OK, sounds good
4. Exponent is a good idea, but I think you'll have a hard time using `**` as the syntax. I think, when I wrote the parser for the calculator language, that I made it so that every operator must have exactly 1 character -- this was much easier and I didn't think we might want double-char operators down the line (but I could be wrong regarding the parser limitation -- or it could be upgraded..). Easy solution is to choose different syntax (1-char-wide).
5. Great idea! An accumulator keyword could be very very useful.
6. OK, this could workout, but note that the mix of colors might make it worse than not having said distinction.
